### PR TITLE
Backport Bitcoin Qt/Gui changes up to 0.14.x part 3

### PR DESCRIPTION
**Pull request by codablock**
This is the third part of Qt/Gui related PRs being backported from Bitcoin to Dash.

I tried to stick with modal overlay related PRs, but also included some PRs which contain changes later touched by the overlay related PRs.

This PR also includes parts of the network activity toggle I mentioned [here](https://github.com/dashpay/dash/pull/1615#issuecomment-328062535). I added a commit on top of that PR to revert the functionality of the network status icon, so that it keeps opening the peers list instead of toggling network state. This means, that for now it's only possible to toggle network status with the new RPC call setnetworkactive. Later we may add a context menu to the status icon which gives an option to toggle network status. At that point we should probably also replace the network disabled icon to something that fits to the style of the Dash network status.

https://github.com/bitcoin/bitcoin/pull/9218 also introduced the ClickableLabel which was then used for the network toggle and the sync spinner icon. While reverting the changes to the network status icon, I decided to also use the ClickableLabel for the status icon (removes the QPushButton flat button + style handling, making future backporting easier).

With this backport PR merged, we have all the functionality of the modal overlay from Bitcoin 0.14. The remaining backports that I'll do are not related to the overlay, and thus can be postponed to later (after the next releast).

### CODE REVIEW

File: src/qt/sendcoinsdialog.cpp [hunk 1]
```diff
@@ -17,6 +17,7 @@
 #include "walletmodel.h"
 
 #include "base58.h"
+#include "chainparams.h"
```
[the remark this answers is not in the document]
Comment (codablock) [reply src/qt/sendcoinsdialog.cpp]: Remainder of https://github.com/bitcoin/bitcoin/pull/9588, which originally also updated parts in sendcoinsdialog.cpp. I omitted these changes for Dash as the whole fee handling in Dash seems to be different. I not added a commit on top to remove the include.

File: contrib/debian/copyright [hunk 2]
```diff
@@ -51,7 +51,10 @@ Comment: Site: https://github.com/stephenhutchings/typicons.font
 
 Files: src/qt/res/icons/connect*.png
        src/qt/res/src/connect-*.svg
+       src/qt/res/icons/network_disabled.png
```
[the remark this answers is not in the document]
Comment (codablock) [reply contrib/debian/copyright]: I noticed that all icons suffer from wrong entries in contrib/debian/copyright and thus decided to not touch this one as well. I you feel this should be fixed, I can put it onto my TODO and later create a PR that fixes it for all icons.

**Comment by codablock** [comment]
Regarding fNetworkActive, if you feel it's too dangerous to include this at this point in time, I can remove the commits completely and figure out the merge conflicts. I will then add a dedicated commit to only this functionality, including a proper fix for the missing UI (context menu) for it.

File: contrib/debian/copyright (hunk 2 above)
[the remark this answers is not in the document]
Comment (UdjinM6) [reply contrib/debian/copyright]: I'm not sure I understand - you moved icons in this exact PR in 0056497196ae6bfa0048d672f71d2a21146ef5d6 so why not fixing references at the very same time? We should be fixing issues, not introducing new ones ;) Other broken references should be fixed later in a separate PR ofc.

**Comment by UdjinM6** [comment]
re `fNetworkActive`:
I have another idea - instead of tracking down every `ConnectNode` we can probably just do this
```diff
diff --git a/src/net.cpp b/src/net.cpp
index d90e1442d..d4a4222a0 100644
--- a/src/net.cpp
+++ b/src/net.cpp
@@ -346,6 +346,10 @@ bool CConnman::CheckIncomingNonce(uint64_t nonce)
 
 CNode* CConnman::ConnectNode(CAddress addrConnect, const char *pszDest, bool fConnectToMasternode)
 {
+    if (!fNetworkActive) {
+        return NULL;
+    }
+
     if (pszDest == NULL) {
         // we clean masternode connections in CMasternodeMan::ProcessMasternodeConnections()
         // so should be safe to skip this and connect to local Hot MN on CActiveMasternode::ManageState()
```
I did a quick test and it seems to be working as expected without any issues.

**Comment by UdjinM6** [comment]
btw, we need this https://github.com/bitcoin/bitcoin/pull/11237 too I guess

File: contrib/debian/copyright (hunk 2 above)
[the remark this answers is not in the document]
Comment (codablock) [reply contrib/debian/copyright]: Ok, fixed this and added a TODO for the other icons.

**Comment by codablock** [comment]
I've added https://github.com/bitcoin/bitcoin/pull/9528 and https://github.com/bitcoin/bitcoin/pull/11237 to the PR. The first one was already on my local branch for the next batch and the second one is pretty new (not even in Bitcoin 0.15).

**Pull request merged**